libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergDialogFragment.kt

```diff
@@ -0,0 +1,129 @@
+package org.wordpress.android.editor.gutenberg
+
+import android.app.Dialog
+import android.content.Context
+import android.content.DialogInterface
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatDialogFragment
+import androidx.fragment.app.Fragment
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+
+class GutenbergDialogFragment() : AppCompatDialogFragment() {
+    private lateinit var mTag: String
```

Comment:
👋 @SiobhyB , only did a quick glance here, and since I know you've taken inspiration from the BasicDialog implementation I wonder if you can elaborate on the choice of having these dialog "properties" (Tag, message, etc) be members of the Fragment while [they are part of a model class over at the BasicDialog implementation](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/posts/BasicDialogViewModel.kt#L38-L45). Not saying there's an issue with this, haven't really checked deep enough yet, just wondering what's the context/insight on this. Thanks!

Reply:
Ah, I used [BasicFragmentDialog.kt](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/posts/BasicFragmentDialog.kt) as the template, which implements the "properties" in this way. I've gone ahead to edit the commit message that incorrectly referenced BasicDialog.kt rather than BasicFragmentDialog.kt, sorry for the confusion!

Reply:
Ahh, I see, thanks for the update!

Reply:
@SiobhyB I have taken a look at the `BasicFragmentDialog.kt` you mentioned as your point of reference and compared it to the `GutenbergDialogFragment` and the main difference I saw was the inclusion of the `mediaId`. I am thinking we could reuse `BasicFragmentDialog.kt` and track the `mediaId` within the `GutenbergDialogFragment` itself. Wdyt?

Reply:
@jd-alexander, the new class also removes the third `setNeutralButton` and adds [the ability for the parent fragment to implement the positive/negative button interfaces](https://github.com/wordpress-mobile/WordPress-Android/pull/14503/files#diff-8889a43bfa80f8522889ae2f69695ac5eda5b29189bb29c9915a9c4825dae6e7R86) (rather than [the host activity](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/posts/BasicFragmentDialog.kt#L83), which is the default with `BasicFragmentDialog.kt` and would require changes to `EditPostActivity` in our case). Would you still lean towards re-using `BasicFragmentDialog.kt` over a new class with those additional changes considered?

Reply:
Ah, thanks for expounding on that. We can go ahead and keep this class then. The only thing I would change is probably the `mediaId` references since the name of the class is `GutenbergDialogFragment ` and there could be other use cases for it in the future. In the interest of time and not overcomplicating the issue we could even remove the `media` and make it just `id` Ideally, I would utilize generics, so we could make it a `param` value that isn't bound to a type and in this case, we could pass the `Int` but yes, we don't have to do that optimization at this time so we could probably evaluate changing the name for now. Let me know what you think.

Reply:
> In the interest of time and not overcomplicating the issue we could even remove the `media` and make it just `id`.

That makes sense! I've gone ahead to implement that change in https://github.com/wordpress-mobile/WordPress-Android/pull/14503/commits/40310c385d7c435912942c2a9b0b2ea6610d1445.

> Ideally, I would utilize generics, so we could make it a param value that isn't bound to a type and in this case, we could pass the Int but yes, we don't have to do that optimization at this time so we could probably evaluate changing the name for now. Let me know what you think.

That sounds great, I've created [an issue to track other places where this class](https://github.com/wordpress-mobile/WordPress-Android/issues/14676) can be used and have made a note of your comments for optimizing it in the comments here: https://github.com/wordpress-mobile/WordPress-Android/issues/14676#issuecomment-852024183. Does iterating on the class as part of that separate issue seem like a reasonable approach to you?

Reply:
Makes sense to iterate in a separate effort, outside this PR 👍